osf/management/commands/check_ia_metadata.py

```diff
@@ -0,0 +1,105 @@
+import re
+import requests
+from django.core.management.base import BaseCommand
+from osf.models import Registration
+from website import settings
+import logging
+from django.db.models import F
+
+
+logger = logging.getLogger(__name__)
+
+
+class IAMetadataError(Exception):
+    def __init__(self, message=None, fields=None):
+        self.message = message
+        self.fields = fields
+
+
+get_ia_field = lambda field: Registration.IA_MAPPED_NAMES.get(field, field)
+
+mirrored_attrs = list(Registration.SYNCED_WITH_IA)
+
+mirrored_fields = mirrored_attrs + ['subjects', 'tags', 'affiliated_institutions']
+mirrored_relationships = [
+    'subjects__text',
+    'affiliated_institutions__name',
+    'tags__name',
+]
+
+
+def check_ia_metadata(collection=settings.IA_ROOT_COLLECTION, guids=None):
+    item_data = requests.get(
+        f'https://archive.org/advancedsearch.php'
+        f'?q=collection%3A({collection}) AND identifier:(osf-registrations-*-{settings.ID_VERSION})'
+        f'&fl[]='
+        + f'&fl[]='.join([get_ia_field(field) for field in mirrored_fields])
+        + f'&fl[]=identifier'
+        f'&rows=100000'
```

Comment:
Yes, we're looking into re-doing the check_metadata functionality in a follow-up ticket. Don't worry though I don't think this will result in a script that checks each of the >10,000 against the >10,000 IA items, this is probably just going to be used for spot checking and small batches to aid QA.